mmcv/ops/focal_loss.py

```diff
@@ -143,7 +142,8 @@ def forward(ctx,
                 weight: Optional[torch.Tensor] = None,
                 reduction='mean') -> torch.Tensor:
 
-        assert isinstance(target, (torch.LongTensor, torch.cuda.LongTensor))
+        assert isinstance(
+            target, (torch.Tensor, torch.LongTensor, torch.cuda.LongTensor))
```

Comment:
Shoud this line be rewrote as `assert target.dtype == torch.long`?

Reply:
https://github.com/open-mmlab/mmcv/pull/2262#discussion_r983464377

Since the class of the long type tensor is different for different devices (for example, cuda is torch.cuda.LongTensor, npu is torch.npu.LongTensor), the verification here is mainly to avoid non-long type tensor input, and I think it is reasonable to check the dtype

Reply:
did it